tsdb/encoding/encoding.go

```diff
@@ -207,11 +208,15 @@ func (d *Decbuf) Varint64() int64 {
 	if d.E != nil {
 		return 0
 	}
-	x, n := binary.Varint(d.B)
+	ux, n := varint.Uvarint(d.B)
 	if n < 1 {
 		d.E = ErrInvalidSize
 		return 0
 	}
+	x := int64(ux >> 1)
+	if ux&1 != 0 {
+		x = ^x
+	}
```

Comment:
Can we add a comment here saying what it is trying to do here?

Reply:
Added.

Reply:
Thanks!